src/jdk.internal.vm.ci/share/classes/jdk/vm/ci/hotspot/HotSpotVMConfigAccess.java

```diff
@@ -315,7 +322,7 @@ private VMField getField(String name, String cppType, boolean required) {
         }
 
         // Make sure the native type is still the type we expect.
-        if (cppType != null && !cppType.equals(entry.type)) {
+        if (cppType != null && !typeEquals(cppType, entry.type)) {
```

Comment:
We have conditional code in Graal (e.g. [here](https://github.com/oracle/graal/blob/5097b1dabf01fd6d2ea1ea3b470060a138d49fa2/compiler/src/jdk.internal.vm.compiler/src/org/graalvm/compiler/hotspot/GraalHotSpotVMConfig.java#L392-L405)) to deal with type changes in vmstructs so there's no need to fix it here. I'd prefer to not have this code do type aliasing like this as it might hide other changes we need to make in Graal.

Reply:
Okay, if you can fix it in Graal, that would be great.  Thanks!